### ISSUE TITLE
[Proposal] Improper documentation of Deprecated Methods in Gym Docs (Typo)

### ISSUE THREAD

**Issue by Mukilan-Krishnakumar** [Farama-Foundation/Gymnasium#133]
### Proposal

I have been going through Gym documentation, learning the basics from the Docs. I did notice a minor inconvenience with the documentation on the [Core page](https://www.gymlibrary.dev/api/core/).  As of `gym` version 0.26, the `done` return has been [deprecated](https://github.com/openai/gym/releases/tag/0.26.0). 

This is not being represented elegantly in the documentation. Documentation looks a bit clunky with this style of expressing deprecation. 

<img width="1434" alt="image" src="https://user-images.githubusercontent.com/71809485/202050700-ade303ed-302b-4fb8-bb47-3818e84926c7.png"> 

This is due to the usage of `.. autofunction:: gymnasium.Env.step` in [Gymnasium/docs/api/env.md](https://github.com/Farama-Foundation/Gymnasium/blob/a13b9c40fe40922e255c1c1997998151a6d1a4f0/docs/api/env.md) 

Sphinx's `autofunction` automatically pulls documentation from the doc-string. In the core.py file of Gym, the docstring is defined as such:
<img width="1418" alt="image" src="https://user-images.githubusercontent.com/71809485/202051755-44d560dc-b6fc-46bf-b815-585f766f7cff.png">

We can make the documentation a bit more clean by using the `..deprecated::` directive. It is clearly defined in their [docs](http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated).

This is only a slight tardiness, still would appreciate it being formatted.  

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

**Comment by pseudo-rnd-thoughts** [comment]
@Mukilan-Krishnakumar Thanks for spotting that, could you make a PR solving this issue?

**Comment by axb2035** [comment]
BTW: The page link in the issue description is to the previous [gymlibrary.dev](https://www.gymlibrary.dev/api/core/) 

The gymlibrary.dev documetnation has been superceeded by [https://gymnasium.farama.org/api/env/](https://gymnasium.farama.org/api/env/)

Noting that the issue exists in new site as well.

**Comment by Mukilan-Krishnakumar** [comment]
> @Mukilan-Krishnakumar Thanks for spotting that, could you make a PR solving this issue?

Yeah, Definitely, this is my first time contributing to open source, so let me figure some things out with PR and stuff. I need a couple of days to work through it. 

Also, should I close this issue?

**Comment by pseudo-rnd-thoughts** [comment]
No, leave the issue open and only close it when the issue is solved

You should be able to download the repo and modify the `gymnasium.core.Env.step` function docstring and we build the website. See the readme.md in docs.
 Then you will be able to test that the fix works
